### DISCUSSION TITLE
AppDeployToolkit and Scheduled Task

### DISCUSSION THREAD

**Discussion by Pollewops** [PSAppDeployToolkit/PSAppDeployToolkit#968] [Show and tell]
Tried to make AppDeployToolkit and Scheduled Task working with a schedules task for a Computer Restart script.
Based on get-pendingreboot the script prompt user to restart device if required weekly.
Problem is that script runs by SYSTEM account and has user interaction required.
Was playing with Deploy-Application.exe and ServiceUIx64.exe but that this not work. The trick was to use Deploy-Application.ps1.

The question is WHY the EXE won't work. What is in it that does make it fail???

But what does work is the following:
- create scheduled task and configure to run as SYSTEM
- run command: ServiceUIx64.exe (for Windows 11 x64 bit is required which can be obtained from Microsoft)
- argument: -process:explorer.exe "C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe" -WindowStyle Hidden -File .\Deploy-Application.ps1
- Folder: describe here the folder in which the ServiceUIx64.exe and Delploy-Application.ps1 are located

With above I do have a successful interactive mechanism working.
When the task runs I only see a small and very quick popup appearring from powershell i think.
It would be great to have the Deploy-Application.EXE working here....so if someone know how then I really am interested in testing it.

Thanks

**Comment by DanGough** [comment]
It should work fine - try the new Invoke-ServiceUI.ps1 script to see if that helps?
